core/domain/skill_services.py

```diff
@@ -1147,3 +1147,74 @@ def filter_skills_by_mastery(user_id, skill_ids):
         if skill_id in filtered_skill_ids:
             arranged_filtered_skill_ids.append(skill_id)
     return arranged_filtered_skill_ids
+
+
+def get_categorized_and_untriaged_skills_dicts(
+    skill_summary_dicts, skill_ids_assigned_to_some_topic, merged_skill_ids):
+    """Fetches the categorized and untriaged skills' descriptions and ids.
+
+    Args:
+        skill_summary_dicts: list(dict). The list of
+            skill summary dicts.
+        skill_ids_assigned_to_some_topic: set(str). The set of skill ids which
+            are assigned to some topic.
+        merged_skill_ids: list(str). List of skill IDs of merged skills.
+
+    Returns:
+        dict, list(dict). The categorized skills dict and list of
+        skill summaries of skills which are not assigned to any topic.
+    """
+    topics = topic_fetchers.get_all_topics()
+
+    untriaged_skill_summary_dicts = []
+    categorized_skills_dict = {}
+
+    skill_ids_and_details = []
+    skill_ids = []
+
+    for topic in topics:
+        subtopics = topic.subtopics
+        categorized_skills_dict[topic.name] = {}
+        categorized_skills_dict[topic.name]['uncategorized'] = []
+        for skill_id in topic.uncategorized_skill_ids:
+            skill_ids.append(skill_id)
+            skill_ids_and_details.append({
+                'skill_id': skill_id,
+                'topic_name': topic.name,
+                'skill_type': 'uncategorized'
+            })
+        for subtopic in subtopics:
+            categorized_skills_dict[topic.name][subtopic.title] = []
+            for skill_id in subtopic.skill_ids:
+                skill_ids.append(skill_id)
+                skill_ids_and_details.append({
+                    'skill_id': skill_id,
+                    'topic_name': topic.name,
+                    'skill_type': 'categorized',
```

Comment:
You don't need this field, just use subtopic_title=None for the uncategorized case. Then you can keep the key structure of all the dicts the same.

Reply:
Done.